Menu.ItemHelpText: better line breaking

himanshupathak95 opened the pull request:
Fixes: #66998 

## What?
Changed CSS word-break property from `break-all` to `break-word` in meta fields for better text readability when content wraps.

## Why?
Currently, text in meta fields breaks incorrectly within words (e.g., "white" breaks as "whit e") due to using `break-all`. This creates poor readability, especially for descriptions and long text content. Using `break-word` ensures text breaks at appropriate word boundaries.

## Testing Instructions
1. Add this test code to create a custom post type with meta fields in `functions.php` in your theme:
```php
function init_test_meta_fields() {
   register_post_type(
       'landmark',
       array(
           'label'        => 'Landmark',
           'public'       => true,
           'supports'     => array( 'title', 'editor', 'comments', 'revisions', 'trackbacks', 'author', 'excerpt', 'page-attributes', 'thumbnail', 'custom-fields', 'post-formats' ),
           'has_archive'  => true,
           'show_in_rest' => true,
       ),
   );
   register_meta(
       'post',
       'landmark_description',
       array(
           'object_subtype' => 'landmark',
           'label'          => 'Description',
           'show_in_rest'   => true,
           'single'         => true,
           'type'           => 'string',
           'default'        => 'Inspired by the Eiffel Tower, this red and white tower is one of Tokyo\'s most recognizable landmarks.',
       )
   );
}

add_action( 'init', 'init_test_meta_fields' );
```

2. Create a new Landmark post
3. Check the attributes >> content desctiption field in meta fields
4. Verify text now breaks correctly at word boundaries
5. Test with various content lengths including URLs and long words

![Your paragraph text](https://github.com/user-attachments/assets/8ce0f88f-e68b-4a70-becf-abf9d9659c8b)


## Screenshots 

Before `word-break: break-all;` : 

![1](https://github.com/user-attachments/assets/0cff0389-278a-4668-8157-00a65ff1335e)


After `word-break: break-word;` : 

![2](https://github.com/user-attachments/assets/3a9b417b-7f64-4afd-b0df-e347bdbb8fcc)

himanshupathak95 committed:
Fix: Use break-word instead of break-all in attribute description

Changed break-all to use word-break for better readability.

juanfra reviewed (commented):
Thanks for your PR @himanshupathak95 

It's testing nicely to me.

https://github.com/user-attachments/assets/2b3ab768-76a5-4149-bf91-0a2c86b740f4

Can you please add an entry on the unreleased section of the [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) for components?

mirka commented on packages/components/src/menu/styles.ts:
```diff
@@ -380,7 +380,7 @@ export const MenuItemHelpText = styled( Truncate )`
 	line-height: 16px;
 	color: ${ LIGHTER_TEXT_COLOR };
-	word-break: break-all;
+	word-break: break-word;
```
`word-break: break-word` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/CSS/word-break#browser_compatibility), so we should use the non-deprecated equivalent:

```suggestion
	overflow-wrap: anywhere;
```

himanshupathak95 replied:
Hey @mirka, Thanks for the suggestion. 

I have made the suggested changes. Please let me know if there is anything else that can be done.

mirka reviewed (commented):
We can see that the original rule was put in place (#63916) with a pretty specific case in mind — a URL with a hyphen in the middle, like `https://m.media-amazon.com/images/M/MV5BOTI5ZTNkYWQtNDg2Mi00MTBmLTliMGItNTI5YWI5OTZkM2Y2XkEyXkFqcGdeQXVyNzU1NzE3NTg@._V1_QL75_UX500_CR0,47,500,281_.jpg`.

I do understand that `break-all` is not acceptable for wrapping (even if it's fine for truncation) in normal text, so I think reverting to `word-break: normal` is sensible.

Though we will see non-ideal wrapping for certain URLs, I think it's an acceptable compromise as long as they don't overflow and break the container, as was the original intent of the `break-all`:

<img width="297" alt="A URL with a hyphen wraps in an non-ideal way" src="https://github.com/user-attachments/assets/aa3120a6-ab9d-48a4-80b4-1383698d8202">

cc @ciampo for a gut check.

himanshupathak95 committed:
CSS: Replace deprecated word-break with overflow-wrap

Replace deprecated `word-break: break-word` with the modern equivalent
`overflow-wrap: anywhere` to handle text wrapping. This change maintains
the same text wrapping behavior while using the current standard CSS
property.

himanshupathak95 committed:
Merge branch 'trunk' into fix/attribute-description-word-break-style

himanshupathak95 commented:
> Can you please add an entry on the unreleased section of the [changelog](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/CHANGELOG.md) for components?

Hey @juanfra, Thanks for replying.

I am not really sure what entry is to be made. I'd really appreciate it if you could steer me in the right direction.

juanfra commented:
> I am not really sure what entry is to be made. I'd really appreciate it if you could steer me in the right direction.

@himanshupathak95 When making changes to WordPress components, it’s important to add a changelog entry in packages/components/CHANGELOG.md. At the top of this file, you’ll find an “Unreleased” section, which is where you should add your entry. This ensures the changes are included in the next release.

Within the “Unreleased” section, there are sub-sections like “Bug Fixes” and “Enhancements.” Add your entry under the most appropriate sub-section, following the standard format:

```
-   `ComponentName`: Description of the change ([#PR-ID](URL-OF-PR)).
```

himanshupathak95 committed:
Docs: Add changelog entry for MenuItemHelpText text wrapping fix

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -14,6 +14,7 @@
 -   `ToggleGroupControl`: Fix active background for `0` value ([#66855](https://github.com/WordPress/gutenberg/pull/66855)).
 -   `SlotFill`: Fix a bug where a stale value of `fillProps` could be used ([#67000](https://github.com/WordPress/gutenberg/pull/67000)).
 -   `ColorPicker`: Add accessible label for copy button ([#67094](https://github.com/WordPress/gutenberg/pull/67094)).
+-   `MenuItemHelpText`: Fix text wrapping to prevent unintended word breaks by using `overflow-wrap: anywhere` ([#67011](https://github.com/WordPress/gutenberg/pull/67011)).
```
Nit: adding a dot to represent the current component name

```suggestion
-   `Menu.ItemHelpText`: Fix text wrapping to prevent unintended word breaks by using `overflow-wrap: anywhere` ([#67011](https://github.com/WordPress/gutenberg/pull/67011)).
```

ciampo commented on packages/components/CHANGELOG.md:
```diff
@@ -14,6 +14,7 @@
 -   `ToggleGroupControl`: Fix active background for `0` value ([#66855](https://github.com/WordPress/gutenberg/pull/66855)).
 -   `SlotFill`: Fix a bug where a stale value of `fillProps` could be used ([#67000](https://github.com/WordPress/gutenberg/pull/67000)).
 -   `ColorPicker`: Add accessible label for copy button ([#67094](https://github.com/WordPress/gutenberg/pull/67094)).
+-   `MenuItemHelpText`: Fix text wrapping to prevent unintended word breaks by using `overflow-wrap: anywhere` ([#67011](https://github.com/WordPress/gutenberg/pull/67011)).
```
Another nit, but I think that the `### Internal` section would be more fitting for this CHANGELOG entry, given that the component is still a private API

himanshupathak95 replied:
Hey @ciampo, Thanks for the suggestions.

I have addressed all the feedback. Please let me know if there is anything else that can be done.

ciampo reviewed (approved):
The fix seems sensible to me, thank you for the ping @mirka !

Left a couple of minor comments, we can merge this PR once they are addressed

himanshupathak95 committed:
Docs: Update changelog entry for Menu.ItemHelpText to internal section

himanshupathak95 committed:
Merge branch 'trunk' into fix/attribute-description-word-break-style

mirka reviewed (approved):
Looks good, thank you for the follow-ups!
